app/src/components/common/list_item/index.tsx

```diff
@@ -136,21 +139,52 @@ const ListItemWrapper: React.FC<ListItemWrapperProps> = (props: ListItemWrapperP
 }
 
 export const ListItem: React.FC<Props> = (props: Props) => {
-  const { data } = props
-  const { address, question, resolution, status } = data
+  const context = useConnectedWeb3Context()
+  const { account, library: provider } = context
+  const [amount, setAmount] = useState('')
+  const [symbol, setSymbol] = useState('')
+
+  const { market } = props
+  const { collateralToken, collateralVolume, id: address, outcomeTokenAmounts } = market
+
+  const { question } = market.conditions[0]
+  const { outcomes, title } = question
+
+  const endsIn = moment(new Date(question.openingTimestamp * 1000)).fromNow()
+
+  useEffect(() => {
+    const setToken = async () => {
+      if (!account) {
+        return
+      }
+      const erc20Service = new ERC20Service(provider, account, collateralToken)
```

Comment:
How does this work? `account` is the current connected address, not an erc20 :confused:

Reply:
`constructor(provider: any, signerAddress: Maybe<string>, tokenAddress: string) {`

The third argument is an erc20, maybe I could pass null on the second argument.

Reply:
Oh, no, my bad, sorry.